cli/responseemitter.go

```diff
@@ -70,26 +74,52 @@ func (re *responseEmitter) SetEncoder(enc func(io.Writer) cmds.Encoder) {
 	re.enc = enc(re.stdout)
 }
 
-func (re *responseEmitter) SetError(v interface{}, errType cmdkit.ErrorType) {
+func (re *responseEmitter) CloseWithError(err error) error {
```

Comment:
It would be really great if this method didnt return an error. I can't think of any situation where the caller wants to do something with the error, or even can. I think it's likely better to just log an error and move on.

Reply:
The reason I let made this return an error is that `CloseWithError(error) error` is idiomatic and is used throughout the entire Go ecosystem. I don't see a reason to diverge - maybe we don't really need it in ipfs, but possible somewhere else. And now that the command's `Run` function just returns an error we don't call this so often that it's too much of a pain for the caller to do the logging themselves.